### ISSUE TITLE
Compute orientation similarity map from the n best near matches per map pixel

### ISSUE THREAD

**Issue by hakonanes** [pyxem/orix#51]
EMsoft provides the orientation similarity metric (OSM). From [their wiki](https://github.com/EMsoft-org/EMsoft/wiki/EMgetOSM):

> [...] a map for which the intensity in each pixel measures how many of the top N matches that pixel has in common with its nearest neighbors. For a well annealed material, neighbouring pixels have very similar top match lists, so the OSM value will be high; near grain boundaries, one or more of the neighbors will have a completely different top match list, so the OSM value will be substantially smaller. As a result, Orientation Similarity maps generally show a clear outline of the microstructural components (grains, phases, voids, etc.).

This should be implemented as a method in the `CrystalMap` class after #49 is merged.

EMsoft implementation: EBSDgetOrientationSimilarityMap (https://github.com/EMsoft-org/EMsoft/blob/f5e6c8a5cb186478f5d34597708d4962f245a794/Source/EMsoftHDFLib/commonmod.f90#L335).

**Comment by dnjohnstone** [comment]
Do you have a reference to the paper in which they described this? I only ask because I'm pretty sure I've read it but I can't find it right now...

**Comment by hakonanes** [comment]
Of course: Marquardt, Katharina, De Graef, Marc, Singh, Saransh, Marquardt, Hauke, Rosenthal, Anja, Koizuimi, Sanae: Quantitative electron backscatter diffraction (EBSD) data analyses using the dictionary indexing (DI) approach: Overcoming indexing difficulties on geological materials , American Mineralogist 102(9), 1843–1855, 2017. DOI: https://doi.org/10.2138/am-2017-6062

**Comment by dnjohnstone** [comment]
Ok right - I was running a microscope with Katharina last week... probably should have known that...

**Comment by hakonanes** [comment]
Wow, there you go! I'm a big fan of that paper.

**Comment by dnjohnstone** [comment]
I think that this remains on the roadmap with relatively high priority, but I also think there are enough other items in the 0.5.0 milestone that are higher priority, therefore assigning to v0.6.0

**Comment by hakonanes** [comment]
Yeah, we want this for assessing pattern comparisons in kikuchipy. But not immediately, so happy to wait.

**Comment by hakonanes** [comment]
The kikuchipy PR https://github.com/pyxem/kikuchipy/pull/234 contains calculation of the orientation similarity map from a 3D map with 2D navigation shape and n number of indices per navigation point, the indices pointing to triplets of Euler angles in a 1D Euler angle array.

It was decided (off GitHub yesterday) that this code would be merged into kikuchipy in the `kikuchipy.indexing` module for now, but should be considered ported to `orix.crystal_map.CrystalMap` in the future.

**Comment by pc494** [comment]
To confirm, I think we will eventually want `CrystalMap` to do this.

**Comment by dnjohnstone** [comment]
> To confirm, I think we will eventually want `CrystalMap` to do this.

Agree

**Comment by hakonanes** [comment]
> > To confirm, I think we will eventually want `CrystalMap` to do this.
> 
> Agree

this is implemented in kikuchipy (kikuchipy.indexing.orientation_similarity_map) by @onatlandsmyr. either i or him and i will move it to orix after we have released a v0.3 of kikuchipy in december.

**Comment by hakonanes** [comment]
We have computation of orientation similarity from indices into a simulation dictionary (as used by e.g. EMsoft) in kikuchipy: https://github.com/pyxem/kikuchipy/blob/master/kikuchipy/indexing/orientation_similarity_map.py. The plan is to move that functionality into a `CrystalMap.add_orientation_similarity()` method (or called something similar) for v0.7.
